src/libcore/num/mod.rs

```diff
@@ -4580,6 +4602,29 @@ impl_from! { u32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0"
 // Float -> Float
 impl_from! { f32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0")] }
 
+// Integer -> Boolean
```

Comment:
This is lossy, so feels questionable whether it should be there in `From`.

Reply:
Updated to use TryFrom instead.